docs/reactivity/integrate-backend-api.md

```diff
@@ -0,0 +1,67 @@
+# Building a reactive frontend with Oryx
+
+## Qualify the request
+
+The components, services and adapters are customizable, mainly by mapping defining a new component implementation or by providing a custom service. To ensure that the request can be customized, a best practice is to use a so-called qualifier in the application flow. The qualifying object can be customized using standard TypeScript.
+
+```ts
+export interface ProductQualifier {
+  sku?: string;
+  include?: string[];
+}
+
+export class DefaultProductService implements ProductService {
+  get(qualifier: ProductQualifier): Observable<NullableGeneric<Product>> {
+    return this.getProductData(qualifier).value;
+  }
+}
+```
+
+The qualifier will travel down through all the application layers. Whenever a custom property must be added, or in case of new qualifying aspects going forward, the object can be extended. The extended object will arrive down, so that the right request can be made in the lowest layer (adapter).
+
+## Designing the data model
+
+In modern web applications, it is common to communicate with an API to retrieve data. However, the response from an API can often be complex and not suitable for direct use in a component. This is where the concept of client models comes into play. A client model is a representation of the data that is specifically tailored for the needs of the client side application.
+
+In the Oryx framework, we use adapters to transform the response of an API into a client model. Adapters are responsible for making HTTP requests to load data from APIs and also provide normalizers (or serializers) to transform the data into a more readable format. This is especially important when working with complex data standards such as JSON-API, as these can be difficult for a component to work with directly.
+
+By transforming the data from an API into a client model, we are able to abstract away the complexities of the API response and provide a simplified, more accessible representation of the data to the components in our application. This allows us to write components that are easy to maintain and understand, while still being able to take advantage of the underlying data provided by the API.
+
+A sequence diagram of this high-level architecture is provided below:
+
+```mermaid
+sequenceDiagram
+autonumber
+
+participant Service
+participant Adapter
+participant Normalizer
+participant API
+
+activate Service
+Service->>Adapter: get()
+activate Adapter
+Adapter->>API: Sends API call
+API-->>Adapter: Returns response
+loop each normalizer
+    Adapter->>Normalizer: Normalizes response
+    Normalizer-->>Adapter: Returns normalized response
+end
+Adapter-->>Service: Returns response
+deactivate Adapter
+deactivate Service
+
+```
+
+The following steps are visualized:
+
+1. Service requests data from Adapter.
```

Comment:
Are there multiple services in an app?

Reply:
Yes, each domain has at least 1 service. For example `ProductService', `CartService`, `SearchService`, etc.